.github/workflows/source-checks.yml

```diff
@@ -43,6 +43,11 @@ jobs:
           ./configure
           make -j $(nproc)
 
+      - name: check formatting with python-black
+        run: |
+          black --version
+          black --check .
```

Comment:
This means PRs/CI will fail if formatting is not exact. I'm okay with that, but it does mean contributors/developers will have to use this EXACT version of black.

I do worry that we'll be chasing availability of black versions. I think it effectively forces everyone to use what's available in pip. It also imposes a bit of maintenance burden to update black occasionally.